pilot/pkg/networking/util/util.go

```diff
@@ -298,7 +298,7 @@ func IsTCPMetadataExchangeEnabled(node *model.Proxy) bool {
 // ConvertLocality converts '/' separated locality string to Locality struct.
 func ConvertLocality(locality string) *core.Locality {
 	if locality == "" {
-		return &core.Locality{}
```

Comment:
?? @hzxuzhonghu is this the cause of errors?

Reply:
This is reverting a fix I made a while back: https://github.com/istio/istio/pull/20094

if we undo this it will break other things

> 
> * Fix for #20077. When there is no
> locality, traffic will never be sent to the pod, even with locality lb
> disabled. The fix here is setting locality to {} instead of nil

Reply:
We can probably find a better way to handle this. The issue with not returning "nil" here is that it's making the following if statement dead code: https://github.com/istio/istio/blob/master/pkg/bootstrap/config.go#L241